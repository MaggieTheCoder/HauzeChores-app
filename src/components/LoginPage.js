import React, { useState } from "react";
import propTypes from "prop-types";
import {
  createUserWithEmailAndPassword,
  signInWithEmailAndPassword,
  onAuthStateChanged,
} from "firebase/auth";
import { auth } from "../firebase-config";
import "../styles/LoginPage.css";
import postNewUser from "../requests/postNewUser";
import Header from "./Header";
import Footer from "./Footer";
import setUserDetailsOnLogin from "../requests/setUserDetailsOnLogin";

const LoginPage = ({ setUser, setHouseId, setUserId, setCode }) => {
  const [signInDetails, setSignInDetails] = useState();
  const [registerDetails, setRegisterDetails] = useState();
  const [errorMessage, setErrorMessage] = useState();

  onAuthStateChanged(auth, (currentUser) => {
    setUser(currentUser);
  });

  const handleSignInInputChange = (e) => {
    setSignInDetails({ ...signInDetails, [e.target.name]: e.target.value });
  };

  const signIn = async (e) => {
    e.preventDefault();
    console.log(signInDetails);
    try {
      const user = await signInWithEmailAndPassword(
        auth,
        signInDetails.email,
        signInDetails.password
      );
      console.log(user);
      setUserDetailsOnLogin(
        signInDetails.email,
        setHouseId,
        setUserId,
        setCode
      );
      e.target.reset();
    } catch (error) {
      console.log(error);
      setErrorMessage(error.message);
    }
  };

  const handleRegisterInputChange = (e) => {
    setRegisterDetails({ ...registerDetails, [e.target.name]: e.target.value });
  };

  const register = async (e) => {
    e.preventDefault();
    console.log(registerDetails);
    try {
      const user = await createUserWithEmailAndPassword(
        auth,
        registerDetails.email,
        registerDetails.password
      );
      console.log(user);
      postNewUser(registerDetails.email);
      e.target.reset();
    } catch (error) {
      console.log(error);
      setErrorMessage(error.message);
    }
  };
  return (
    <>
      <Header />
      <div className="Login">
        <form onSubmit={signIn} className="Login__login">
          <input
            className="email__input__box"
            type="email"
            placeholder="email"
            name="email"
            onChange={handleSignInInputChange}
            data-testid="login-email"
          />
          <input
            className="password__input__box"
            type="password"
            placeholder="password"
            name="password"
            onChange={handleSignInInputChange}
            data-testid="login-password"
          />
          <button type="submit" className="Login__button">
            LOG IN
          </button>
        </form>
        <hr />
        <form onSubmit={register} className="Login__register">
          <input
            className="email__input__box"
            type="email"
            placeholder="email"
            name="email"
            onChange={handleRegisterInputChange}
            data-testid="register-email"
          />
          <input
            className="password__input__box"
            type="password"
            placeholder="password"
            name="password"
            onChange={handleRegisterInputChange}
            data-testid="register-password"
          />
          <button type="submit" className="Login__button">
            REGISTER
          </button>
        </form>
        {errorMessage && <p>Error! {errorMessage}</p>}
      </div>
      <Footer />
    </>
  );
};

export default LoginPage;

LoginPage.propTypes = {
  setUser: propTypes.func.isRequired,
  setHouseId: propTypes.func.isRequired,
  setUserId: propTypes.func.isRequired,
  setCode: propTypes.func.isRequired,
};
