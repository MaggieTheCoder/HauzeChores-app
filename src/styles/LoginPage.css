

.Login {
  padding-top:100px;
  padding-bottom: 15px;
  display: flex;
  flex-direction: column;
  align-items: center;
  width : 100%;
  height : 700px;
  background-color: #99E5FF;
  border-radius : 90px;
}

.Login__login {
  width: 50%;
  display: flex;
  flex-direction: column;
  margin-bottom: 30px;
}

.Login__register {
  width: 50%;
  display: flex;
  flex-direction: column;
}

.email__input__box {

  font-family: "Menlo", sans-serif;
  font-size: 25px;
  font-style: italic;
  margin-bottom: 15px;
  padding: 5px;
  border-radius: 15px;
  text-align: center;
}
.password__input__box {

  font-family: "Menlo", sans-serif;
  font-size: 25px;
  font-style: italic;
  margin-bottom: 15px;
  padding: 5px;
  border-radius: 15px;
  text-align: center;
}

.Login__button {
  font-family: "Menlo", sans-serif;
  font-size: 25px;
  margin-bottom: 10px;
  background-color: #C60982;
  border-radius: 15px;
  padding: 5px;
  color: white;
}