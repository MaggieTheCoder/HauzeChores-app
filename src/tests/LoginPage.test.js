import React from "react";
import { render, screen } from "@testing-library/react";
import LoginPage from "../components/LoginPage";

describe("LoginPage component", () => {
  const validProps = {
    setUser: () => {},
  };
  test("", () => {
    const { asFragment } = render(<LoginPage setUser={validProps.setUser} />);

    expect(asFragment()).toMatchSnapshot();
  });

  test("Login form renders correctly", () => {
    render(<LoginPage setUser={validProps.setUser} />);
    const loginButton = screen.getByText("Login");
    const loginEmailInput = screen.getByTestId("login-email");
    const loginPasswordInput = screen.getByTestId("login-password");

    expect(loginButton).toBeInTheDocument();
    expect(loginEmailInput).toBeInTheDocument();
    expect(loginPasswordInput).toBeInTheDocument();
  });

  test("Register form renders correctly", () => {
    render(<LoginPage setUser={validProps.setUser} />);
    const registerButton = screen.getByText("Register");
    const registerEmailInput = screen.getByTestId("register-email");
    const registerPasswordInput = screen.getByTestId("register-password");

    expect(registerButton).toBeInTheDocument();
    expect(registerEmailInput).toBeInTheDocument();
    expect(registerPasswordInput).toBeInTheDocument();
  });
});
