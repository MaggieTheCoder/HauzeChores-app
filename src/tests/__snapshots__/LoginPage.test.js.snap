// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginPage component  1`] = `
<DocumentFragment>
  <div
    class="Login"
  >
    <form
      class="Login__login"
    >
      <input
        data-testid="login-email"
        name="email"
        placeholder="email"
        type="email"
      />
      <input
        data-testid="login-password"
        name="password"
        placeholder="password"
        type="password"
      />
      <button
        class="Login__button"
        type="submit"
      >
        Login
      </button>
    </form>
    <hr />
    <form
      class="Login__register"
    >
      <input
        data-testid="register-email"
        name="email"
        placeholder="email"
        type="email"
      />
      <input
        data-testid="register-password"
        name="password"
        placeholder="password"
        type="password"
      />
      <button
        class="Login__button"
        type="submit"
      >
        Register
      </button>
    </form>
  </div>
</DocumentFragment>
`;
